loader: Fixes for map creation from daemon

pchaigno opened the pull request:
This pull request introduces a few follow-up fixes for #10626. Please see commit messages.

pchaigno committed:
loader: Ignore cilium_call_policy symbol in ELF

The policy call map was renamed, but the list of symbols to skip in the
ELF wasn't updated, resulting in the following warning:

    2020-03-25T23:08:21.073661843Z level=warning msg="Skipping symbol substitution" subsys=elf symbol=cilium_call_policy

Fixes: 5d6b669 ("maps/policymap: Rename policy call map to clarify intent")
Signed-off-by: Paul Chaignon <paul@cilium.io>

pchaigno committed:
pkg/neighborsmap: Missing padding for map values

Commit 8fd7415 introduces a Go definition of the NodePort neighbor maps,
to enable their creation during the daemon's initialization. The value
structure is however missing some padding to match its C counterpart,
resulting in the following warnings:

    2020-03-25T23:07:55.245972761Z level=warning msg="Value-size mismatch for BPF map" file-path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 new=6 old=8 subsys=bpf
    2020-03-25T23:07:55.245975572Z level=warning msg="Removing map to allow for property upgrade (expect map data loss)" file-path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
    2020-03-25T23:07:55.256431723Z level=debug msg="Registered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf
    2020-03-25T23:07:55.256461108Z level=debug msg="Unregistered BPF map" path=/sys/fs/bpf/tc/globals/cilium_nodeport_neigh4 subsys=bpf

Fixes: 8fd7415 ("maps/neighborsmap, daemon: Create map from daemon")
Signed-off-by: Paul Chaignon <paul@cilium.io>

borkmann reviewed (approved):
LGTM, thanks Paul!

brb commented on pkg/maps/neighborsmap/neighborsmap.go:
```diff
@@ -49,6 +49,7 @@ type Key6 struct {
 // +k8s:deepcopy-gen:interfaces=github.com/cilium/cilium/pkg/bpf.MapValue
 type Value struct {
```
To avoid the {key,value}-size mismatch problems in the future, I think it makes sense to pass this and others recently introduced Go structs to the alignchecker.

pchaigno replied:
Uh, I'm pretty sure I had done that at some point in #10626. I must have removed those changes by mistake when I reorganized the commit history to bisect :-/

Anyway, I have added checks for the three new maps (neighborsmap, eventsmap, and signalmap) and double checked that other maps are already tested. I also reorganized the commits of the present pull request locally to make sure that the alignchecker detects the missing padding (by moving the fix before the alignchecker update).

pchaigno committed:
alignchecker: check eventsmap, neighborsmap, and signalmap

Three new maps were added to pkg/maps/ in #10626. The corresponding
checks in alignchecker are however missing. This commit adds them.

Signed-off-by: Paul Chaignon <paul@cilium.io>
